edx_lint/files/common_constraints.txt

```diff
@@ -21,7 +21,3 @@ elasticsearch<7.14.0
 
 # django-simple-history>3.0.0 adds indexing and causes a lot of migrations to be affected
 django-simple-history==3.0.0
-
-# tox>4.0.0 isn't yet compatible with many tox plugins, causing CI failures in almost all repos.
```

Comment:
> ... causing CI failures in almost all repos.

This scares me, and seem like a good reason to have a common constraint. Did you confirm that a repo that was failing on 4.0.0 now works with the latest upgrade, and that we can reasonably expect that it is now passing for most repos?

Reply:
Unfortunately, it's nearly impossible to test this because of how the common constraints are setup in a repo.  You can't actually remove the constraint at the repo level without `make upgrade` automatically adding it back before it rebuilds requirements.

That said, this could be added to the constraints at the repo level and my plan was to do a big announce so that maintainers are prepared to do that if they see issues.  As I said in the description, most of our repos have `tox-battery` as a dependency and that already constrains `tox<4.0.0` so I'm less concerned about this breaking everything like it did when tox 4.0.0 was first released. (At that time tox-battery did not have its constraint on `tox<4.0.0`.

Reply:
We could use this hacky solution on a branch just for testing: https://github.com/openedx/edx-platform/commit/56d12bf230bc1fc5ec930c3afdf412ba2cae7239. I think it would be useful. If we discover a problem that is easily fixable, but that all or many repos might run into, it might be nice to provide the solution rather than having everyone add constraints, and then having the same problem we have now (a constraint where we don't want one), only a much more global version of it. Thoughts?

Reply:
Also see https://github.com/openedx/edx-lint/issues/355 which I just moved from Jira: https://github.com/openedx/edx-lint/issues/355.

Reply:
@robrap good call on digging further into this. I had missed the fact that 0.6.2 of tox-battery(the version that constrains tox) has been merged but has not been released.  I think the real path forward is gonna have to be dropping the dependency on tox-battery everywhere first and then dropping this common constraint.

Reply:
Cool. Thanks for digging further. Good luck. Should this ticket be closed for now, or will you block on some new issue?

Reply:
Sounds good. Are you planning on ticketing and pushing forward the `tox-battery` work? Was this all just inform for me at this point? Thanks.